src/content.ts

```diff
@@ -1,69 +1,121 @@
-// import * as config from "../config/config";
-// import * as storage from "../storage/storage";
-import { ActivityPubOpts } from "./core/activityPub/activityPub";
-import { ConfigOpts } from "./config";
-import { NotImplementedError } from "./core/utilities/errors";
-import { BatchBroadcastMessage, BatchReactionMessage, BatchReplyMessage } from "./core/batch/batchMesssages";
+import * as activityPub from "./core/activityPub/activityPub";
+import * as batchMessages from "./core/batch/batchMesssages";
+import * as config from "./config";
+import * as messages from "./core/messages/messages";
+import * as storage from "./core/storage/storage";
+import { getRandomString, validateDSNPId, MissingUser } from "./core/utilities";
 
 /**
- * broadcast() creates a broadcast DSNP event and enqueues it for the next
- * batch. This method is not yet implemented.
+ * InvalidActivityPubOpts represents an error in the activity pub options
+ * provided to a method.
+ */
+export const InvalidActivityPubOpts = new Error("Invalid activity pub options.");
+
+/**
+ * InvalidInReplyTo represents an error in the DSNP Id provided for the
+ * inReplyTo parameter.
+ */
+export const InvalidInReplyTo = new Error("Invalid DSNP Id for inReplyTo");
+
+/**
+ * broadcast() creates an activity pub file with the given content options,
+ * uploads it with a random filename using the configured storage adapter and
+ * creates a DSNP broadcast message for the hosted file for later announcement.
+ *
+ * @throws {@link MissingSigner}
+ * Thrown if the signer is not configured.
  *
- * @param content The content for the note to broadcast
- * @param opts    Optional. Configuration overrides, such as from address, if any
- * @return The Batch Broadcast Message
+ * @param contentOptions Options for the activity pub content to generate
+ * @param opts           Optional. Configuration overrides, such as from address, if any
+ * @returns              A Signed DSNP Broadcast message ready for inclusion in a batch
  */
-export const broadcast = async (_content: ActivityPubOpts, _opts?: ConfigOpts): Promise<BatchBroadcastMessage> => {
-  throw NotImplementedError;
-
-  // const config = config.getConfig(opts);
-  //
-  // const activityPubObject = activityPub.create(activityPubOpts);
-  // const activityPubHash = activityPub.hash(activityPubObject);
-  // const uri = storage.store(activityPubObject, opts);
-  //
-  // return events.createBroadcastEvent(uri, activityPubHash);
+export const broadcast = async (
+  contentOptions: activityPub.ActivityPubOpts,
+  opts?: config.ConfigOpts
+): Promise<batchMessages.BatchBroadcastMessage> => {
+  // Create the activity pub file
```

Comment:
I'm a little conflicted. Given that this is a public project which may have outside contributors and alternative implementations in the future, I can see comments being valuable, but I can also see your point.

@acruikshank @shannonwells, any opinions?

Reply:
I am not against adding comments when needed. But in this case I do not think they are adding much value.
For example,

// Get current user id
const { currentUserId } = await config.getConfig(opts); 
if (!currentUserId) throw MissingUser;

I think that your code here is pretty clear. An alternative way to remove that comment would be to write a function called getUserId?

Reply:
Since we've decided to add typedoc style comments, then we should use them everywhere that there are functions and objects exposed to SDK users,  even if it seems like it's not needed.  For one thing, if we do generate separate documentation, then not having the docs is surprising and makes the docs look incomplete.  I'm all for self-documenting code, but we're documenting this for outside developers and IMO consistency and completeness are more important. Comments do no harm to the code.

Reply:
I'm in favor of keeping the typedoc on the function, but I think @enddynayn is referring to the internal comments. I think he's right that we should only be adding internal comments when we feel there's the potential for confusion in the code. Comments contribute to the maintenance costs of the codebase, and they need be adding some useful information if order to offset that cost.